shell/utils/inactivity.ts

```diff
@@ -0,0 +1,183 @@
+
+          return data;
+        }
+
+        console.error(`Could not find any user activity for token ${ sessionTokenName }`); // eslint-disable-line no-console
+
+        return defaultParsedTTLData;
+      } catch (e) {
+        console.error(`Could not update user activity for token ${ sessionTokenName }`, e); // eslint-disable-line no-console
+
+        return defaultParsedTTLData;
+      }
+    }
+  }
+
+  console.error('Cannot find a userActivity for this session OR this user does not have permissions to list the necessary resources'); // eslint-disable-line no-console
+
+  return defaultParsedTTLData;
+}
+
+export async function checkUserActivityData(store: any, sessionTokenName: string): Promise<userActivityResponse> {
+  try {
+    const updatedData = await store.dispatch('management/find', { type: EXT.USERACTIVITY, id: sessionTokenName });
+
+    // update stored data
+    storedUserActivityResponse = updatedData;
+
+    return Promise.resolve(updatedData);
+  } catch (e) {
+    console.error(`Could not GET UserActivity for session token ${ sessionTokenName }`, e); // eslint-disable-line no-console
+
+    return Promise.reject(e);
+  }
+}
+
+export async function updateUserActivityToken(store: any, sessionTokenName: string):Promise<userActivityResponse> {
+  const updateUserActivity = await store.dispatch('management/create', {
+    apiVersion: 'ext.cattle.io/v1',
+    kind:       'UserActivity',
+    type:       EXT.USERACTIVITY,
+    metadata:   { name: sessionTokenName },
+    spec:       { tokenId: sessionTokenName }
+  });
+
+  try {
+    const savedData = await updateUserActivity.save();
+
+    // update stored data
+    storedUserActivityResponse = savedData;
+
+    return Promise.resolve(savedData);
+  } catch (e) {
+    console.error(`Could not update (POST) UserActivity for session token ${ sessionTokenName }`, e); // eslint-disable-line no-console
+
+    return Promise.reject(e);
+  }
+}
+
+export function parseTTLData(userActivityData: userActivityResponse): parsedInactivitySetting {
+  const currDate = Date.now();
+  const endDate = new Date(userActivityData.status?.expiresAt).getTime();
+
+  // let's give this a 3 second buffer so that we can make sure the logout happens by the frontend
+  const thresholdSeconds = Math.floor((endDate - currDate) / 1000) - 3;
+
+  // Amount of time the user sees the inactivity warning
+  const courtesyTimerVal = Math.floor(thresholdSeconds * 0.1);
+  const courtesyTimer = Math.min(courtesyTimerVal, 60 * 5); // Never show the modal more than 5 minutes
+
+  const courtesyCountdown = courtesyTimer;
+
+  // Amount of time before the user sees the inactivity warning
+  // Note - time before warning is shown + time warning is shown = settings threshold (total amount of time)
+  const showModalAfter = thresholdSeconds - courtesyTimer;
+
+  return {
+    enabled:          true,
```

Comment:
I don't see enabled used anywhere. Can this be removed?

Reply:
it's used on the `inactivity` component, `mounted` method, `if (backendInactivityData.enabled)`.

It's considered enabled if `auth-user-session-idle-ttl-minutes` is less than `auth-user-session-ttl-minutes`

Reply:
I am going to refactor a bit the logic to make it reactive to `auth-user-session-idle-ttl-minutes` and `userActivity`, which with that make the `enabled` even more important. I'll resolve this for now and push the changes once they are ready